spark/src/main/scala/ai/chronon/spark/stats/ConsistencyJob.scala

```diff
@@ -88,10 +88,18 @@ class ConsistencyJob(session: SparkSession, joinConf: Join, endDate: String) ext
     if (unfilledRanges.isEmpty) return null
     val allMetrics = unfilledRanges.map { unfilled =>
       val comparisonDf = tableUtils.sql(unfilled.genScanQuery(null, joinConf.metaData.comparisonTable))
+      // External parts / contextual features don't get logged in the online data but do appear in the comparison table.
```

Comment:
I think the root cause description might be wrong. 

External / contextual parts do get logged in the online data and show up in the log table. They also show up in the comparison table but the values will be NULL. 

However, currently `CompareBaseJob.compare` will only compare columns that appear both on the left and on the right. In the current code, we pass in `loggedDfNoExternalCols` to `CompareBaseJob.compare`, and therefore already remove the external / contextual parts. So IIUC the current code shouldn't fail for this scenario.